packages/styled-components/transforms/src/visitors/transpile_css_prop/top_level_binding_collector.rs

```diff
@@ -0,0 +1,100 @@
+use swc_atoms::JsWord;
+use swc_common::collections::AHashSet;
+use swc_ecmascript::{
+    ast::{
+        ArrowExpr, ClassDecl, FnDecl, Function, ImportDefaultSpecifier, ImportNamedSpecifier,
+        ImportStarAsSpecifier, ModuleItem, ObjectPatProp, Param, Pat, Stmt, VarDeclarator,
+    },
+    visit::{noop_visit_type, Visit, VisitWith},
+};
+
+// Modified from swc_ecma_utils/src/lib.rs:BindingCollector.
+pub struct TopLevelBindingCollector {
+    bindings: AHashSet<JsWord>,
+}
+
+impl TopLevelBindingCollector {
+    fn add(&mut self, i: &JsWord) {
+        self.bindings.insert(i.clone());
+    }
+}
+
+impl Visit for TopLevelBindingCollector {
+    noop_visit_type!();
+
+    fn visit_class_decl(&mut self, node: &ClassDecl) {
+        self.add(&node.ident.sym);
+    }
+
+    fn visit_fn_decl(&mut self, node: &FnDecl) {
+        self.add(&node.ident.sym);
+    }
+
+    fn visit_pat(&mut self, node: &Pat) {
```

Comment:
This will not handle `const [ foo = default ] = init;`
The LHS of `Pat::Assign` should be visited.